server/aesdsocket.c

```diff
@@ -0,0 +1,284 @@
+/**
+ * A simple socket server to accept incoming connection
+ * and echo received data back to client.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <syslog.h>
+#include <signal.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#define DEBUG
+#ifdef DEBUG
+#define SYSLOG_OPTIONS      (LOG_PERROR | LOG_NDELAY)
+#else
+#define SYSLOG_OPTIONS      (LOG_NDELAY)
+#endif
+
+#define MAX_ARGS            1
+#define PORT                "9000"      // Port we're listening on
+#define BACKLOG             10
+#define MAX_LEN             100
+
+volatile sig_atomic_t interrupted = 0;
+const char *log_file = "/var/tmp/aesdsocketdata";
+
+static void signal_handler(int signo)
+{
+    interrupted = 1;
+    syslog(LOG_INFO, "Caught signal, exiting");
+}
+
+static int echo_data(int client)
+{
+    char *buf;
+    int fd, ret;
+    struct stat statbuf;
+
+    fd = open(log_file, O_RDONLY, (S_IWUSR|S_IRUSR|S_IRGRP|S_IROTH));
+    if (fd < 0) {
+        syslog(LOG_ERR, "Unable to open file %s", log_file);
+        return -1;
+    }
+
+    fstat(fd, &statbuf);
+    if ((buf = (char *)calloc(statbuf.st_size, sizeof(char))) == NULL) {
```

Comment:
@hgajapathy what assumption are you making here about the total size of all packets sent and the size of the heap?